firebase-firestore/src/test/java/com/google/firebase/firestore/core/FilterTest.java

```diff
@@ -85,4 +86,11 @@ public void testCompositeFilterNestedChecks() {
     assertFalse(orFilter2.isFlat());
     assertFalse(orFilter2.isFlatConjunction());
   }
+
+  @Test
+  public void testCanonicalIdOfFlatConjunctions() {
+    Target target1 = query("col").filter(A).filter(B).filter(C).toTarget();
+    Target target2 = query("col").filter(andFilters(A, B, C)).toTarget();
+    assertEquals(target1.getCanonicalId(), target2.getCanonicalId());
```

Comment:
Why we choose to compare target instead of query? IMO CanonicalId for query is most of users would choose to use?

Reply:
Even though Query `getCanonicalId` calls the Target `getCanonicalId` under the hood, you're right that it's more future-proof to test the Query API directly. I've posted 2 PRs (one for Android and one for Web) to update this.